File: examples/projects/benchmarks/matmul/matmul.cpp
// benchmark: 256x256 f64 矩阵乘法 —— 与 matmul.zeta 同逻辑
#include <cstdio>
#include <vector>

int main() {
    const int n = 256;
    const int total = n * n;
    std::vector<double> a(total, 1.0001), b(total, 1.0001), c(total, 0.0);
    for (int x = 0; x < n; x++) {
        for (int y = 0; y < n; y++) {
            double s = 0.0;
            for (int k = 0; k < n; k++) s += a[x * n + k] * b[k * n + y];
            c[x * n + y] = s;
        }
    }
    double sum = 0.0;
    for (int i = 0; i < total; i++) sum += c[i];
    printf("%f\n", sum);
    return 0;
}

File: examples/projects/benchmarks/matmul/matmul.rs
// benchmark: 256x256 f64 矩阵乘法 —— 与 matmul.zeta 同逻辑
fn main() {
    let n = 256;
    let total = n * n;
    let mut a = vec![1.0001; total];
    let mut b = vec![1.0001; total];
    let mut c = vec![0.0; total];
    let mut x = 0;
    while x < n {
        let mut y = 0;
        while y < n {
            let mut s = 0.0;
            let mut k = 0;
            while k < n {
                s = s + a[x * n + k] * b[k * n + y];
                k = k + 1;
            }
            c[x * n + y] = s;
            y = y + 1;
        }
        x = x + 1;
    }
    let mut sum = 0.0;
    let mut idx = 0;
    while idx < total {
        sum = sum + c[idx];
        idx = idx + 1;
    }
    println!("{}", sum);
}
